core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowResource.scala

```diff
@@ -563,4 +607,132 @@ class WorkflowResource {
+    * @return The workflowID filter.
+    */
+  def getWorkflowIdFilter(workflowIDs: java.util.List[UInteger]): Condition = {
+    var workflowIdFilter: Condition = noCondition()
+    if (workflowIDs != null && !workflowIDs.isEmpty) {
+      for (workflowID <- workflowIDs) {
+        workflowIdFilter = workflowIdFilter.or(WORKFLOW.WID.eq(workflowID))
+      }
+    }
+    workflowIdFilter
+  }
+
+  /**
+    * Helper function to retrieve the creation date filter.
+    * Applies a filter based on the specified creation start and end dates.
+    *
+    * @param creationStartDate The start date to filter by.
+    * @param creationEndDate   The end date to filter by.
+    * @return The creation date filter.
+    */
+  def getCreationDateFilter(
+      creationStartDate: String,
+      creationEndDate: String
+  ): Condition = {
+    var creationDateFilter: Condition = noCondition()
+    if (creationStartDate.nonEmpty || creationEndDate.nonEmpty) {
+      val start = if (creationStartDate.nonEmpty) creationStartDate else "1970-01-01"
+      val end = if (creationEndDate.nonEmpty) creationEndDate else "9999-12-31"
+      val dateFormat = new SimpleDateFormat("yyyy-MM-dd")
+      val startTimestamp = new Timestamp(dateFormat.parse(start).getTime)
+      val endTimestamp = new Timestamp(
+        dateFormat.parse(end).getTime + TimeUnit.DAYS.toMillis(1) - 1
+      )
+      creationDateFilter = WORKFLOW.CREATION_TIME.between(startTimestamp, endTimestamp)
+    }
+    creationDateFilter
+  }
+
+  /**
+    * Helper function to retrieve the modified date filter.
+    * Applies a filter based on the specified modified start and end dates.
+    * @param modifiedStartDate The modified start date to filter by.
+    * @param modifiedEndDate   The modified end date to filter by.
+    * @return The modified date filter.
+    */
+  def getModifiedDateFilter(modifiedStartDate: String, modifiedEndDate: String): Condition = {
+    var modifiedDateFilter: Condition = noCondition()
+    if (modifiedStartDate.nonEmpty || modifiedEndDate.nonEmpty) {
+      val start = if (modifiedStartDate.nonEmpty) modifiedStartDate else "1970-01-01"
+      val end = if (modifiedEndDate.nonEmpty) modifiedEndDate else "9999-12-31"
+      val dateFormat = new SimpleDateFormat("yyyy-MM-dd")
+      val startTimestamp = new Timestamp(dateFormat.parse(start).getTime)
+      val endTimestamp = new Timestamp(
+        dateFormat.parse(end).getTime + TimeUnit.DAYS.toMillis(1) - 1
+      )
+      modifiedDateFilter = WORKFLOW.LAST_MODIFIED_TIME.between(startTimestamp, endTimestamp)
+    }
+    modifiedDateFilter
+  }
+
+  /**
+    * Helper function to retrieve the operators filter.
+    * Applies a filter based on the specified operators.
+    *
+    * @param operators The list of operators to filter by.
+    * @return The operators filter.
+    */
+  def getOperatorsFilter(operators: java.util.List[String]): Condition = {
+    var operatorsFilter: Condition = noCondition()
```

Comment:
I don't expect an array to be passed in. Each query parameter should be a single operator. The split and the removal of [] can be deleted. 
Also, only the exact operator type will be passed in. So no need to use LIKE.

Reply:
Could you elaborate on this? From my understanding, if there are more than one operator in the request like `/workflow/search?operator=PostgreSQLSource&operator=MySQLSource`
This query will be converted to an array of string(operators).
Could you also elaborate on your new implementation idea for this function? Thanks!

Reply:
From my understanding of the code, this is also trying to parse requests like `/workflow/search?operator=[PostgreSQLSource, MySQLSource]`, which doesn't really need to be a valid format

Reply:
what do you suggest if we don't use LIKE

Reply:
It can be an exact match. i.e. `=`

Reply:
Oh I see what it's doing here. I misunderstood the code.
This is searching for the operator in the `content` column. 
Since we already have the full-text index, why don't we use that?

Reply:
But if this is simply keeping what the old code is doing, we don't have to do this improvement in this PR.

Reply:
> Oh I see what it's doing here. I misunderstood the code. This is searching for the operator in the `content` column. Since we already have the full-text index, why don't we use that?
Actually, this would require a new full-text index, something like this workflow(content).

Reply:
> But if this is simply keeping what the old code is doing, we don't have to do this improvement in this PR.

Yea, I suggest we keep this in a separate PR. 
But, I do think we should remove  this line` operator.replaceAllLiterally(" ", "").stripPrefix("[").stripSuffix("]").split(',')`